frontend/e-buddy/src/components/chatbot/ChatPreview.tsx

```
@@ -48,184 +241,499 @@ export const ChatPreview = ({ customization, onClose }: ChatPreviewProps) => {
+
+  // Render different message content based on type
+  const renderMessageContent = (content: MessageContent) => {
+    if (typeof content === "string") {
+      return <div className="whitespace-pre-wrap">{content}</div>;
+    }
+
+    // For complex content types
+    switch (content.type) {
+      case "product_card":
+        const product = content.products?.[0];
+        if (!product) return <div>{content.text}</div>;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)